server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Rights.scala

```diff
@@ -20,36 +20,65 @@
 package org.apache.james.jmap.mail
 
 import org.apache.james.core.Username
-import org.apache.james.mailbox.model.MailboxACL
-import org.apache.james.mailbox.model.MailboxACL.{EntryKey, Rfc4314Rights}
+import org.apache.james.mailbox.model.{MailboxACL => JavaMailboxACL}
+import org.apache.james.mailbox.model.MailboxACL.{EntryKey, Rfc4314Rights => JavaRfc4314Rights, Right => JavaRight}
 import org.slf4j.{Logger, LoggerFactory}
 
-import scala.collection.mutable
 import scala.jdk.CollectionConverters._
 
+object Rfc4314Rights {
+  def fromJava(rights: JavaRfc4314Rights) = Rfc4314Rights(rights.list().asScala.toSeq)
+
+  def fromRights(seq: Seq[Right]): Rfc4314Rights = Rfc4314Rights(seq.map(_.right))
+}
+
+case class Rfc4314Rights(rights: Seq[JavaRight]) {
```

Comment:
Why not use a Set instead of a Seq ?

Reply:
Set is unordered. I see issue within integration tests ahead. I'd rather keep things ordered for now.